core-java-modules/core-java-streams-5/src/test/java/com/baeldung/aggregateEx/AggregateExceptionHandlerUnitTest.java

```diff
@@ -0,0 +1,180 @@
+    }
+
+    @Test
+    public void whenFoundNonInts_thenAggregateExceptionAndReturnOutput() {
+        String[] strings = {"1", "2", "3", "a", "b", "c"};
+        Map resultMap = Arrays.stream(strings)
+                .map(AggregateExceptionHandlerUnitTest::transform)
+                .collect(Collectors.partitioningBy(o -> o instanceof RuntimeException, Collectors.toList()));
+        RuntimeException ex = null;
+        if (resultMap.containsKey(Boolean.TRUE)) {
+            List<RuntimeException> exs = (List<RuntimeException>) resultMap.get(Boolean.TRUE);
+            ex = exs.stream()
+                    .reduce(
+                            new RuntimeException("Errors Occurred"), (o1, o2) -> {
+                                o1.addSuppressed(o2);
+                                return o1;
+                            });
+        }
+        assertEquals("Errors Occurred", ex.getMessage());
+        assertEquals(3, ex.getSuppressed().length);
+    }
+
+    @Test
+    public void givenWrapFunction_whenFoundNonInts_thenAggregateException() throws ExceptionAggregator {
+        String[] strings = {"1", "2", "3", "a", "b", "c"};
+        Map<Boolean, List<Result<Integer>>> resultmap = Arrays.stream(strings)
+                .map(CustomMapper.mapper(Integer::parseInt))
+                .collect(Collectors.partitioningBy(r -> r.getException().isEmpty(), Collectors.toList()));
+
+        if (resultmap.containsKey(Boolean.FALSE)) {
+            List<Result<Integer>> resultList = resultmap.get(Boolean.FALSE);
+            List<Exception> exceptionList = resultList.stream()
+                    .map(opex -> opex.getException().get())
+                    .collect(Collectors.toList());
+
+            assertThrows(ExceptionAggregator.class, () -> handleExceptions(exceptionList));
+        }
+    }
+
+    private void handleExceptions(List<Exception> exceptions) throws ExceptionAggregator {
+        ExceptionAggregator exceptionAggregator = new ExceptionAggregator("Errors occurred");
+        exceptionAggregator.addExceptions(exceptions);
+        throw exceptionAggregator;
+    }
+
+    @Test
+    public void givenExCollector_whenFoundNonInts_thenAggregateException() throws ExceptionAggregator {
+        String[] strings = {"1", "2", "3", "a", "b", "c"};
+        ExceptionCollector exCollector = Arrays.stream(strings)
+                .collect(ExceptionCollector.of(Integer::parseInt));
+        List<Integer> output = exCollector.getResults();
+        List<RuntimeException> runEx = exCollector.getExceptions();
+        assertEquals(3, runEx.size());
+    }
+
+    private static Either<RuntimeException, Integer> processAndReturnEither(String str) {
+        Either<RuntimeException, Integer> either = null;
+        try {
+            either = Either.right(Integer.parseInt(str));
+        } catch (NumberFormatException e) {
+            either = Either.left(new RuntimeException(e));
+        }
+        return either;
```

Comment:
This can be simplified, less code, easier to read:
```suggestion
        try {
            return Either.right(Integer.parseInt(str));
        } catch (NumberFormatException e) {
            return Either.left(new RuntimeException(e));
        }
```

Reply:
agreed